cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -1668,5 +1706,36 @@ pub fn define(
         }
     }
 
+    // Reference type instructions
+
+    // Null references implemented as iconst 0
+    e.enc32(null.bind_ref(R32), rec_pu_id_ref.opcodes(vec![0xb8]));
+
+    e.enc64(null.bind_ref(R32), rec_pu_id_ref.rex().opcodes(vec![0xb8]));
+    e.enc64(null.bind_ref(R32), rec_pu_id_ref.opcodes(vec![0xb8]));
+
+    e.enc64(null.bind_ref(R64), rec_pu_id_ref.rex().opcodes(vec![0xb8]));
+    e.enc64(null.bind_ref(R64), rec_pu_id_ref.opcodes(vec![0xb8]));
+
+    // Sign-extended 32-bit immediate.
+    e.enc64(
+        null.bind_ref(R64),
+        rec_u_id_ref.rex().opcodes(vec![0xc7]).rrr(0).w(),
+    );
+
+    // Finally, the 0xb8 opcode takes an 8-byte immediate with a REX.W prefix.
+    e.enc64(
+        null.bind_ref(R64),
+        rec_pu_iq_ref.opcodes(vec![0xb8]).rex().w(),
```

Comment:
There are 4 different encodings for `null.r64`, which are all unguarded (at the recipe or encoding level), so I think some of them are unreachable. Could we remove the largest ones?

Reply:
Redundant encodings have now been removed.